hangman/lib/hangman/game.ex

```diff
@@ -0,0 +1,164 @@
+defmodule Hangman.Game do
+    @moduledoc """
+    Hangman game implementation.
+    """
+
+    ### Game state struct ###
+
+    alias Hangman.Game
+    defstruct(
+        # DEFAULTS
+        status:     :initializing,
+        guessed:    %MapSet{},
+        turns_left: 7,
+
+        # PLACEHOLDERS (values will be determined during game)
+        word:       "N/A", 
+        last_guess: "N/A"
+    )
+
+    ### Public Functions (Called from API) ###
+
+    @spec new_game(binary()) :: struct()
+    def new_game(word) do 
+        %Game{ 
+            word: word |> String.downcase() 
+        }
+    end
+
+    @spec new_game() :: struct()
```

Comment:
Typespecs!! +3 to you

Just a note on these: rather than using the generic `struct` type, you can actually use %Game{}, which gives you better type checking.

I don't know if you ran dialyzer on your code, but it passes type checks with no warnings.